cluster-operator/src/main/java/io/strimzi/operator/cluster/model/quotas/QuotasPluginUtils.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.quotas;
+
+import io.strimzi.api.kafka.model.common.Condition;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPlugin;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginKafka;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginStrimzi;
+import io.strimzi.operator.cluster.model.KafkaConfiguration;
+import io.strimzi.operator.common.model.StatusUtils;
+import org.apache.kafka.common.quota.ClientQuotaAlteration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class containing methods for handling the configuration around {@link QuotasPlugin}
+ */
+public class QuotasPluginUtils {
+
+    /**
+     * Creates list of {@link ClientQuotaAlteration.Op} based on the configuration in {@link QuotasPluginKafka}.
+     * This list will be then passed to the Admin client and the default user quotas will be set.
+     *
+     * @param quotasPluginKafka     configuration of the quotas
+     * @return  list of {@link ClientQuotaAlteration.Op}
+     */
+    public static List<ClientQuotaAlteration.Op> getListOfClientQuotaAlternationOps(QuotasPluginKafka quotasPluginKafka) {
+        List<ClientQuotaAlteration.Op> ops = new ArrayList<>();
+
+        ops.add(new ClientQuotaAlteration.Op("producer_byte_rate",
+            quotasPluginKafka.getProducerByteRate() != null ? Double.valueOf(quotasPluginKafka.getProducerByteRate()) : null));
+        ops.add(new ClientQuotaAlteration.Op("consumer_byte_rate",
+            quotasPluginKafka.getConsumerByteRate() != null ? Double.valueOf(quotasPluginKafka.getConsumerByteRate()) : null));
+        ops.add(new ClientQuotaAlteration.Op("request_percentage",
+            quotasPluginKafka.getRequestPercentage() != null ? Double.valueOf(quotasPluginKafka.getRequestPercentage()) : null));
+        ops.add(new ClientQuotaAlteration.Op("controller_mutation_rate",
+            quotasPluginKafka.getControllerMutationRate() != null ? quotasPluginKafka.getControllerMutationRate() : null));
+
+        return ops;
+    }
+
+    /**
+     * Checks whether only one of `minAvailableBytesPerVolume` and `minAvailableRatioPerVolume` is configured.
+     * If both, adds a new warning to the list of warnings and configuration of quotas is skipped.
+     *
+     * @param quotasPlugin  quotas plugin configuration
+     * @param warnings      list of warnings where we can put new one in case of need
+     * @return  result of the check
+     */
+    public static boolean isQuotasSpecCorrect(QuotasPlugin quotasPlugin, List<Condition> warnings) {
```

Comment:
Doens't the CRD validate this?

Reply:
I think yes. But in MM2 spec I found that it's anyway checked for  the fields labeled with `OneOf`. I  will check it. Thanks

Reply:
So it seems that the `OneOf` annotation is not "applied" for the sub-types of the classes (meaning it is skipped from `QuotasPluginStrimzi`, but if we specify it in `QuotasPlugin`, it will work).

I removed this method from where, but I added a check for it into the `KafkaCluster.validateConfigurationOfQuotasPlugin()`.